src/gsy_e/gsy_e_core/sim_results/endpoint_buffer.py

```diff
@@ -325,24 +327,24 @@ def update_stats(self, area: "Area", simulation_status: str,
             "percentage_completed": int(progress_info.percentage_completed)
         }
 
-        self.results_handler.update(
-            self.area_result_dict, self.flattened_area_core_stats_dict,
-            self.current_market_time_slot_str
-        )
+        if calculate_results:
+            self.results_handler.update(
+                self.area_result_dict, self.flattened_area_core_stats_dict,
+                self.current_market_time_slot_str)
 
-        self.bids_offers_trades.clear()
-
-        if (ConstSettings.GeneralSettings.EXPORT_OFFER_BID_TRADE_HR or
-                ConstSettings.GeneralSettings.EXPORT_ENERGY_TRADE_PROFILE_HR):
-            self.offer_bid_trade_hr.update(area)
+            if (ConstSettings.GeneralSettings.EXPORT_OFFER_BID_TRADE_HR or
+                    ConstSettings.GeneralSettings.EXPORT_ENERGY_TRADE_PROFILE_HR):
+                self.offer_bid_trade_hr.update(area)
 
         self.result_area_uuids = set()
         self.update_results_area_uuids(area)
-        self.update_offer_bid_trade()
 
-    def update_offer_bid_trade(self) -> None:
+        self._update_offer_bid_trade()
+
+    def _update_offer_bid_trade(self) -> None:
         """Populate self.bids_offers_trades with results from flattened_area_core_stats_dict
         (for local export of statistics)."""
+        self.bids_offers_trades.clear()
```

Comment:
So basically we weren't clearing these before, did you find that by stumbling upon a bug or is this just a precaution (I'm afraid we need to be careful here)?

Reply:
we need to clear the `bids_offers_trades` dict buffer from the last markets' bids, offers, trades before populating these with the new market results again. Have a look at master now, this step is done before calling _update_offer_bid_trade. I just moved it inside the method.

Reply:
I see, in this case all good.